pkg/controller/association/reconciler.go

```diff
@@ -202,7 +202,7 @@ func (r *Reconciler) Reconcile(ctx context.Context, request reconcile.Request) (
 	for _, association := range associations {
 		newStatus, err := r.reconcileAssociation(ctx, association)
 		if err != nil {
-			results.WithError(err)
+			results.WithResult(defaultRequeue).WithError(err)
```

Comment:
nit: why do we need `WithResult(defaultRequeue)` ?

Reply:
This acts as a retry mechanism if association reconciliation failed when requesting an unmanaged resource to get its version.

Reply:
I believe the request is already retried if there's an error, also `RequeueAfter` [is ignored in that case](https://github.com/kubernetes-sigs/controller-runtime/blob/c46b410039fdc9d620a4217bc7c63c5a7fffae46/pkg/internal/controller/controller.go#L320-L344):

```golang
	result, err := c.Reconcile(ctx, req)
	switch {
	case err != nil:
		c.Queue.AddRateLimited(req)
		ctrlmetrics.ReconcileErrors.WithLabelValues(c.Name).Inc()
		ctrlmetrics.ReconcileTotal.WithLabelValues(c.Name, labelError).Inc()
		log.Error(err, "Reconciler error")
	case result.RequeueAfter > 0:
		// The result.RequeueAfter request will be lost, if it is returned
		// along with a non-nil error. But this is intended as
		// We need to drive to stable reconcile loops before queuing due
		// to result.RequestAfter
		c.Queue.Forget(obj)
		c.Queue.AddAfter(req, result.RequeueAfter)
		ctrlmetrics.ReconcileTotal.WithLabelValues(c.Name, labelRequeueAfter).Inc()
	case result.Requeue:
		c.Queue.AddRateLimited(req)
		ctrlmetrics.ReconcileTotal.WithLabelValues(c.Name, labelRequeue).Inc()
	default:
		// Finally, if no error occurs we Forget this item so it does not
		// get queued again until another change happens.
		c.Queue.Forget(obj)
		ctrlmetrics.ReconcileTotal.WithLabelValues(c.Name, labelSuccess).Inc()
	}
```

Reply:
Double-checked and you're absolutely right -> 5f7e6d5.